sql/hive-thriftserver/v1.2.1/src/main/java/org/apache/hive/service/cli/ColumnValue.java

```diff
@@ -195,7 +195,7 @@ public static TColumnValue toTColumnValue(Type type, Object value) {
     case DECIMAL_TYPE:
       return stringValue(((HiveDecimal)value));
     case BINARY_TYPE:
-      return stringValue((String)value);
+      return stringValue(new String((byte[])value));
```

Comment:
We stored it as `Array[Byte]`:
https://github.com/apache/spark/blob/687dd4eb55739f802692b3c5457618fd6558e538/sql/hive-thriftserver/src/main/scala/org/apache/spark/sql/hive/thriftserver/SparkExecuteStatementOperation.scala#L105-L106